neo.UnitTests/UT_ContractManifest.cs

```diff
@@ -0,0 +1,88 @@
+            check.Permissions = new[]
+            {
+                new ContractPermission()
+                {
+                    Contract = ContractPermissionDescriptor.Create(UInt160.Zero),
+                    Methods = WildCardContainer<string>.Create("method1", "method2")
+                }
+            };
+            Assert.AreEqual(manifest.ToString(), check.ToString());
+        }
+
+        [TestMethod]
+        public void ParseFromJson_SafeMethods()
+        {
+            var json = @"{""groups"":[],""features"":{""storage"":false,""payable"":false},""abi"":{""hash"":""0x0000000000000000000000000000000000000000"",""entryPoint"":{""name"":""Main"",""parameters"":[{""name"":""operation"",""type"":""String""},{""name"":""args"",""type"":""Array""}],""returnType"":""Any""},""methods"":[],""events"":[]},""permissions"":[{""contract"":""*"",""methods"":""*""}],""trusts"":[],""safeMethods"":[""balanceOf""]}";
+            var manifest = ContractManifest.Parse(json);
+            Assert.AreEqual(manifest.ToString(), json);
+
+            var check = ContractManifest.CreateDefault(UInt160.Zero);
+            check.SafeMethods = WildCardContainer<string>.Create("balanceOf");
+            Assert.AreEqual(manifest.ToString(), check.ToString());
+        }
+
+        [TestMethod]
+        public void ParseFromJson_Trust()
+        {
+            var json = @"{""groups"":[],""features"":{""storage"":false,""payable"":false},""abi"":{""hash"":""0x0000000000000000000000000000000000000000"",""entryPoint"":{""name"":""Main"",""parameters"":[{""name"":""operation"",""type"":""String""},{""name"":""args"",""type"":""Array""}],""returnType"":""Any""},""methods"":[],""events"":[]},""permissions"":[{""contract"":""*"",""methods"":""*""}],""trusts"":[""0x0000000000000000000000000000000000000001""],""safeMethods"":[]}";
+            var manifest = ContractManifest.Parse(json);
+            Assert.AreEqual(manifest.ToString(), json);
+
+            var check = ContractManifest.CreateDefault(UInt160.Zero);
+            check.Trusts = WildCardContainer<UInt160>.Create(UInt160.Parse("0x0000000000000000000000000000000000000001"));
+            Assert.AreEqual(manifest.ToString(), check.ToString());
+        }
+
+        [TestMethod]
+        public void ParseFromJson_Groups()
+        {
+            var json = @"{""groups"":[{""pubKey"":""03b209fd4f53a7170ea4444e0cb0a6bb6a53c2bd016926989cf85f9b0fba17a70c"",""signature"":""41414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141414141""}],""features"":{""storage"":false,""payable"":false},""abi"":{""hash"":""0x0000000000000000000000000000000000000000"",""entryPoint"":{""name"":""Main"",""parameters"":[{""name"":""operation"",""type"":""String""},{""name"":""args"",""type"":""Array""}],""returnType"":""Any""},""methods"":[],""events"":[]},""permissions"":[{""contract"":""*"",""methods"":""*""}],""trusts"":[],""safeMethods"":[]}";
```

Comment:
How does these groups really work? A contract can be part of more than one group in this current design, right?
And why the signature?

Reply:
Without signature, you can join any group, which is not by design.

Reply:
Ok, I got it, in the current design, how is able to sign for entering? Is it a multisig with 1 single threshold?

Reply:
Just put the signature in the json.

Reply:
The group identifier is the PublicKey, right? `03b209fd4f53a7170ea4444e0cb0a6bb6a53c2bd016926989cf85f9b0fba17a70c`

Which signatures will be allowed? 
If we have 10 members of the group, should we create a MultiSig 1/10? For each new member, append an additional PubKey?

Sorry for the trivial questions, @erikzhang, trying to get the idea here...aehauheau

Reply:
Please check the specification here: #287

> Where pubkey represents the public key of the group and signature is the signature of the contract hash.